src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -343,9 +413,25 @@ void setNotifier(Notifier* n) {
 */
 DBClientBase* getMongoConnection(void)
 {
-    return connection;
+#ifdef UNIT_TEST
+  return connection;
+#else
+  return mongoPoolConnectionGet();
+#endif
 }
 
+DBClientBase* mongoInitialConnectionGetForUnitTest(void)
+{
+  return mongoPoolConnectionGet();
+}
+
+void releaseMongoConnection(DBClientBase* connection)
```

Comment:
After looking at it a little more in-depth, the problem lies in that testInit.cpp/setupDatabase gets a connection by calling mongoInitialConnectionGetForUnitTest, but there is no corresponding function to release the connection.

Creating an issue about this problem.